gst/nnstreamer/tensor_transform/tensor_transform.c

```diff
@@ -1156,30 +1155,21 @@ gst_tensor_transform_arithmetic (GstTensorTransform * filter,
 
 #ifdef HAVE_ORC
   if (orc_supported (filter)) {
-    uint8_t *srcptr = (uint8_t *) inptr;
-
     walk = filter->operators;
-    op_s = (tensor_transform_operator_s *) walk->data;
 
-    if (op_s->op == GTT_OP_TYPECAST) {
-      /**
-       * Typecast should be called at the first.
-       * Do the typecast. If in/out type is same, this will copy the input array to output.
-       */
-      orc_typecast (inptr, outptr, num, in_tensor_type, out_tensor_type);
-      srcptr = outptr;
-
-      walk = g_slist_next (walk);
-    }
+    /**
+     * Typecast should be called at the first.
+     * Do the typecast. If in/out type is same, this will copy the input array to output.
+     */
+    orc_typecast (inptr, outptr, num, in_tensor_type, out_tensor_type);
 
     while (walk) {
       op_s = (tensor_transform_operator_s *) walk->data;
 
       if (op_s->op != GTT_OP_TYPECAST) {
         gst_tensor_transform_typecast_value (filter, &op_s->value,
             out_tensor_type);
-        orc_operator (srcptr, outptr, num, &op_s->value, op_s->op);
-        srcptr = outptr;
+        orc_operator (outptr, num, &op_s->value, op_s->op);
```

Comment:
This might be outside the scope of this PR.
But I think the code in case of `#define HAVE_ORC` and otherwise are inconsistent.

In case of `#define HAVE_ORC`, the first data element of the `GList walk` is always transformed. Then only, operators which are not of type `GTT_OP_TYPECAST` are processed, others are ignored. 
However, without `#define HAVE_ORC`, `GTT_OP_TYPECAST` type is supported for all the data elements in `GList *walk`.

Reply:
@kparichay Without orc, tensor-transform uses ```tensor_transform_operand_s``` to handle operators. (set tensor_transform_operand_s value from input ptr > handle operators > put value to output ptr). 
With orc, we have to pass an array pointer to orc functions, to handle operators. (add, mul, and so on) So, we have to handle typecast or copy data to output array at the first.

About GTT_OP_TYPECAST, tensor-transform does not allow typecast option in the middle of operators. (Please see the function gst_tensor_transform_set_option_data(), REGEX_ARITH_OPTION_TYPECAST to parse the options.)

Reply:
This was because of the mem-allocation during the arith options. If allowing typecast in the middle, tensor-transform should have temporal memories to pass array ptr to ORC functions. For the simplicity, I restricted typecast option only in the first. (Actually, without ORC option, it was OK to allow typecast in the middle of options.)

Reply:
I understand now. Thanks for the explanation.